data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionService.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import com.amazon.dataprepper.model.event.Event;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import javax.inject.Named;
+
+@Named
+class ParseTreeCoercionService {
+    public Object coercePrimaryTerminalNode(final TerminalNode node, final Event event) throws ExpressionCoercionException {
+        final int nodeType = node.getSymbol().getType();
+        final String nodeStringValue = node.getText();
+        switch (nodeType) {
+            case DataPrepperExpressionParser.EscapedJsonPointer:
+                return event.get(nodeStringValue.substring(1, nodeStringValue.length() - 1), Object.class);
```

Comment:
nit: reading the code here I understand what is happening but not why. It might be a good idea to include a variable or create a private function.

```Java
final String jsonPointerWithoutQuotes = nodeStringValue.substring(1, nodeStringValue.length() - 1);
// or
private String removeQuotesFrom(final String text) {
   return nodeStringValue.substring(1, nodeStringValue.length() - 1)
}
```